controllers/statusreport/statusreport_adapter.go

```diff
@@ -86,17 +90,41 @@ func (a *Adapter) EnsureSnapshotTestStatusReportedToGitProvider() (controller.Op
 			return controller.RequeueWithError(err)
 		}
 	}
+	integrationTestScenarios, err := a.loader.GetRequiredIntegrationTestScenariosForApplication(a.client, a.context, a.application)
+	if err != nil {
+		return controller.RequeueWithError(err)
+	}
 
-	if gitops.IsSnapshotMarkedAsPassed(a.snapshot) || gitops.IsSnapshotMarkedAsFailed(a.snapshot) || gitops.IsSnapshotMarkedAsInvalid(a.snapshot) {
-		a.logger.Info("Status has been reported successfully, now in the process of removing the finalizer")
-		// Remove the finalizer from all the Integration PLRs related to the Snapshot
-		err = helpers.RemoveFinalizerFromAllIntegrationPipelineRunsOfSnapshot(a.client, a.logger, a.context, *a.snapshot, helpers.IntegrationPipelineRunFinalizer)
-		if err != nil {
-			return controller.RequeueWithError(err)
-		}
-		a.logger.Info("Successfully removed the finalizer from all the Integration PLRs related to current snapshot", "snapshot.Name", a.snapshot.Name)
+	testStatuses, err := gitops.NewSnapshotIntegrationTestStatusesFromSnapshot(a.snapshot)
+	if err != nil {
+		return controller.RequeueWithError(err)
 	}
 
+	for _, integrationTestScenario := range *integrationTestScenarios {
+		integrationTestScenario := integrationTestScenario // G601
+		testDetails, ok := testStatuses.GetScenarioStatus(integrationTestScenario.Name)
+		if !ok || helpers.IsPipelineRunInFinalState(testDetails) {
+			pipelineRunName := integrationTestScenario.Name
```

Comment:
I think we should get the pipelineRunName from [testDetails.TestPipelineRunName](https://github.com/redhat-appstudio/integration-service/blob/93f0ff0a0ef11d491d249180f8380eba1d9abcce/pkg/integrationteststatus/integration_test_status.go#L97) here instead.

Reply:
yes, also please note, that not every state has PLRname defined, you have to check it for non-emptiness.

Reply:
Maybe you don't need to check status at all, just check if TestPipelineRunName is defined

Reply:
> Maybe you don't need to check status at all, just check if TestPipelineRunName is defined

I think we should still check the pipelineRun status in one way or another so we don't remove finalizers from running ones.

Reply:
ah yes :D ooops